gap/attributes/properties.gi

```diff
@@ -610,12 +615,21 @@ function(S)
   if HasIdempotentGeneratedSubsemigroup(S) or not IsActingSemigroup(S) then
     T := IdempotentGeneratedSubsemigroup(S);
   else
-    min := MinimumList(List(gens, x -> ActionRank(S)(x)));
-    new := Filtered(Idempotents(S), x -> ActionRank(S)(x) >= min);
+    ranks := List(gens, x -> ActionRank(S)(x));
+    min := MinimumList(ranks);
+    if HasIdempotents(S) then
+      new := Filtered(Idempotents(S), x -> ActionRank(S)(x) >= min);
+    else
+      max := MaximumList(ranks);
+      new := [];
+      for i in [min .. max] do
+        Append(new, Idempotents(S, i));
+      od;
+    fi;
     if new = [] then
       return false;
     fi;
-    T := Semigroup(new);
+    T := Semigroup(new, rec(acting := true));
```

Comment:
Ok, I still don't really see why this is necessary, can you explain? Won't `T` automatically have `acting := true` because the `S` does?